File: templates/problems/15_greedy/07_partition_labels/partition_labels_test.go
package partition_labels

import (
	"reflect"
	"testing"
)

func TestPartitionLabels(t *testing.T) {
	tests := []struct {
		name     string
		s        string
		expected []int
	}{
		{"example 1", "ababcbacadefegdehijhklij", []int{9, 7, 8}},
		{"single char", "a", []int{1}},
		{"all same", "aaaa", []int{4}},
		{"all unique", "abcdef", []int{1, 1, 1, 1, 1, 1}},
		{"two partitions", "aabbb", []int{2, 3}},
		{"example 2", "eccbbbbdec", []int{10}},
		{"interleaved", "abac", []int{3, 1}},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := partitionLabels(tt.s)
			if !reflect.DeepEqual(got, tt.expected) {
				t.Errorf("partitionLabels(%q) = %v, want %v", tt.s, got, tt.expected)
			}
		})
	}
}

File: templates/problems/15_greedy/07_partition_labels/typescript/partitionLabels.test.ts
import { describe, it, expect } from 'vitest';
import { partitionLabels } from './partitionLabels';

describe('PartitionLabels', () => {
    it('example 1', () => {
        expect(partitionLabels('ababcbacadefegdehijhklij')).toEqual([9, 7, 8]);
    });

    it('single char', () => {
        expect(partitionLabels('a')).toEqual([1]);
    });

    it('all same', () => {
        expect(partitionLabels('aaaa')).toEqual([4]);
    });

    it('all unique', () => {
        expect(partitionLabels('abcdef')).toEqual([1, 1, 1, 1, 1, 1]);
    });

    it('two partitions', () => {
        expect(partitionLabels('aabbb')).toEqual([2, 3]);
    });

    it('example 2', () => {
        expect(partitionLabels('eccbbbbdec')).toEqual([10]);
    });

    it('interleaved', () => {
        expect(partitionLabels('abac')).toEqual([3, 1]);
    });
});
